pkg/apis/marketplace/v1alpha1/catalogsourceconfig_types.go

```diff
@@ -53,3 +59,36 @@ func (csc *CatalogSourceConfig) EnsureGVK() {
 	}
 	csc.SetGroupVersionKind(gvk)
 }
+
+// RemoveFinalizer removes the operator source finalizer from the
+// OperatorSource ObjectMeta.
+func (csc *CatalogSourceConfig) RemoveFinalizer() {
+	outFinalizers := make([]string, 0)
+	for _, finalizer := range csc.ObjectMeta.Finalizers {
+		if finalizer == CSCFinalizer {
+			continue
+		}
+		outFinalizers = append(outFinalizers, finalizer)
+	}
+
+	csc.ObjectMeta.Finalizers = outFinalizers
+
+	return
+}
+
+// EnsureFinalizer ensures that the operator source finalizer is included
+// in the ObjectMeta Finalizers slice. If it already exists, no state change occurs.
+// If it doesn't, the finalizer is appended to the slice.
+func (csc *CatalogSourceConfig) EnsureFinalizer() {
+	// First check if the opsrc finalizer is already included in the object.
+	for _, finalizer := range csc.ObjectMeta.Finalizers {
+		if finalizer == Finalizer {
+			return
+		}
+	}
+
+	// If it doesn't exist, append the finalizer to the object meta.
+	csc.ObjectMeta.Finalizers = append(csc.ObjectMeta.Finalizers, CSCFinalizer)
+
+	return
+}
```

Comment:
Suggestion: Ensure and Remove functions for both type have the same logic except they act on different finalizer, It might be a good idea to make them common functions.
```
func RemoveFinalizer(objectMeta *metav1.ObjectMeta, finalizer string) {
}

func EnsureFinalizer(objectMeta *metav1.ObjectMeta, finalizer string) {
}
```

And then, each type's Ensure and Delete calls the above
```
func (csc *CatalogSourceConfig) RemoveFinalizer() {
   RemoveFinalizer(&csc.ObjectMeta, CSCFinalizer)
}
```

Reply:
Sure, makes sense.

Reply:
I would say this is a required changed :-) I don't want any repeated code if it can be avoided.